proof/invariant-abstract/ARM_HYP/ArchInvariants_AI.thy

```diff
@@ -2418,17 +2418,14 @@ lemma valid_tcb_arch_ref_lift:
   "tcb_arch_ref t = tcb_arch_ref t' \<Longrightarrow> valid_arch_tcb (tcb_arch t) = valid_arch_tcb (tcb_arch t')"
   by (simp add: valid_arch_tcb_def tcb_arch_ref_def)
 
-lemma valid_arch_tcb_context_update[simp]:
-  "valid_arch_tcb (tcb_context_update f t) = valid_arch_tcb t"
-  unfolding valid_arch_tcb_def obj_at_def by simp
+lemma valid_arch_tcb_simps[simp]:
+  "\<And>f. valid_arch_tcb (tcb_context_update f t) = valid_arch_tcb t"
```

Comment:
do these meta-foralls do anything? (same for `valid_arch_tcb_context_simps` below)

Reply:
It's a pattern I've been using for when I collect a whole bunch of simp rules into one lemma. For example, see the AARCH64 version of this lemma
https://github.com/seL4/l4v/blob/78ad44e4a5594df14a4e50fe17680df2f1eeb4e0/proof/invariant-abstract/AARCH64/ArchInvariants_AI.thy#L1567-L1569
There it's needed because `f` has a different type for each update.
In this case where it's just a single simp rule then it's not needed and I probably did it due to copy-pasting from AARCH64. I could remove it but I sort of like leaving it in, both for more consistency between architectures and to make it easier to modify if this architecture has more fields added in the future.

Reply:
It is consistent with the lemmas just below where it is required, so I'd be fine with leaving it for this one as well.